### PR TITLE
adds `pytest-vcr` to tests

### PR DESCRIPTION
**Pull request by ayushanand18**
## Overview
As per discussion in #120, this PR adds `pytest-vcr` and recorded cassettes of network interaction while running tests.

## Changes Introduced
1. Marks all tests with `vcr` so that all network interactions are recorded.
2. The recorded cassettes are also committed.
3. Adds `pytest-vcr` as a dev dependency.
4. Adds flag to not record while executing GitHub Actions workflow.

Thanks!

### CODE REVIEW

**Review by 7yl4r — APPROVED** [review]

**Comment by ayushanand18** [comment]
@ocefpaf waiting for your feedback on this :)

**Comment by ocefpaf** [comment]
> @ocefpaf waiting for your feedback on this :)

Apologies. I'll review it right now!

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -30,4 +30,4 @@ jobs:
       shell: bash -l {0}
       run: |
         micromamba activate TEST
-        python -m pytest -rxs --cov=pyobis ./pyobis
+        python -m pytest -rxs --cov=pyobis ./pyobis --vcr-record=none
```
Comment (ocefpaf) [inline .github/workflows/tests.yml:33]: I like to add this comment for my future self and future contributors. 
No strong opinion if you do not want to add that here and/or if you want to have it in a dev documentation elsewhere.

```suggestion
        # use --vcr-record=all when recording new cassettes. We never record on CIs though
        python -m pytest -rxs --cov=pyobis ./pyobis --vcr-record=none
```

**Comment by ayushanand18** [comment]
> Apologies. I'll review it right now!

No issues. I just thought you forgot to add a review.

**Review by ocefpaf — APPROVED** [review]
Looks good! Thanks @ayushanand18!!

File: .github/workflows/tests.yml (hunk 1 above)
Comment (ayushanand18) [reply .github/workflows/tests.yml:33]: Thank you so much for the suggestion! This is a valuable addition and I think it makes the new change more understandable without needing to search for the commit messages or this PR.

**Comment by ocefpaf** [comment]
All green! I don't want to merge this b/c there are some commits I added later to fix the CIs, so a third party review would be nice.

Thanks @ayushanand18!

**Comment by ayushanand18** [comment]
> All green! I don't want to merge this b/c there are some commits I added later to fix the CIs, so a third party review would be nice.

Thank you so much! The best thing I like about PRs is I get to learn a lot from code reviews. I worked at a place for some days where code reviews weren't a serious thing and I felt really terrible - this is the reason I love open source ♥️.

I think @7yl4r is the best person to merge this.

**Pull request merged**